policies.go

```diff
@@ -156,6 +156,14 @@ type RetryPolicy interface {
 	GetRetryType(error) RetryType
 }
 
+// This is a similar interface to RetryPolicy
+// If a query is recognized as LWT query it, and its RetryPolicy satisfies this
+// interface, then this interface will be used instead of RetryPolicy.
+type LWTRetryPolicy interface {
```

Comment:
Since `RetryableQuery` and `ExecutableQuery` are tightly connected to each other, I would suggest to add `IsLWT() bool` to `RetryableQuery` and change `RetryPolicy.GetRetryType` to `GetRetryType(RetryableQuery, error) RetryType`

Yes, it is going to be a breaking change, but very light in a sence of fixing it and if user is using built-in retrypolicy it is going to be autoresolved.

WDYT?

Reply:
I am not sure about doing breaking changes. It is a high priority issue and we want to be able to merge it and release as quickly as possible

Reply:
I don't know what is the policy about breaking changes in gocql, because I don't usually work on this driver and I'm not a maintainer. If you want me to I can implement proposed changes. 
My thinking with avoiding breaking changes was:
- Afaik Golang itself maintains backwards compatibility, so I assume its expected in the libraries too
- Even without breaking changes customers often stay on older version. If updating requires changes it will be harder to move users to current version
- It's not just incompatibility with our old version, but with upstream gocql too, so it makes migration harder (and possibly merges with upstream too)
- If it is acceptable to introduce a breaking change, then we can do that in the future too, removing `LWTRetryPolicy` and applying your proposed change to `RetryPolicy`.